### PR TITLE
@types/node: stream.pipeline() should accept iterators since Node 13.10

### PR DESCRIPTION
**Pull request by voxpelli**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/stream.html#stream_stream_pipeline_source_transforms_destination_callback
- [x] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~
- [x] ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.~~

### CODE REVIEW

**Review by a-tarasyuk — APPROVED** [review]

**Review by rbuckton — CHANGES_REQUESTED** [review]
The NodeJS documentation here is poor and there are some inconsistencies between the above type and the actual NodeJS runtime semantics.

File: types/node/stream.d.ts [hunk 1]
```diff
@@ -297,7 +297,11 @@ declare module "stream" {
             function __promisify__(stream: NodeJS.ReadableStream | NodeJS.WritableStream | NodeJS.ReadWriteStream, options?: FinishedOptions): Promise<void>;
         }
 
-        function pipeline<T extends NodeJS.WritableStream>(stream1: NodeJS.ReadableStream, stream2: T, callback?: (err: NodeJS.ErrnoException | null) => void): T;
+        function pipeline<T extends NodeJS.WritableStream>(
+            stream1: NodeJS.ReadableStream | Iterable<any> | AsyncIterable<any>,
```
[the remark this answers is not in the document]
Comment (voxpelli) [reply types/node/stream.d.ts:301]: @rbuckton Something you want me to do meanwhile?

[the remark this answers is not in the document]
Comment (voxpelli) [reply types/node/stream.d.ts:301]: @rbuckton I see that https://github.com/nodejs/node/issues/36437 was merged, how do we move forward with this then?

[the remark this answers is not in the document]
Comment (rbuckton) [reply types/node/stream.d.ts:301]: That's a good question. They merged a fix to NodeJS, but that hasn't shipped yet, as far as I can tell. We either need to wait until it ships (and ensure the package version comment matches the major.minor of the shipping release), or you can make the proposed changes above now, and a separate PR that removes the `function pipeline(streams: readonly any[], ...args: any[]): never;` overload (with the package version comment change) at a later date.

**Comment by voxpelli** [comment]
https://github.com/nodejs/node/issues/36437 is now released as https://github.com/nodejs/node/releases/tag/v15.5.0 @rbuckton, should we push this change into https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48981 and/or continue here?

**Comment by voxpelli** [comment]
I submitted a request to backport the patch to Node 14: https://github.com/nodejs/node/pull/36831

**Comment by voxpelli** [comment]
The backport has been merged, so once it’s released we can merge this I believe @rbuckton

**Comment by voxpelli** [comment]
`v14.15.5` has now been released: https://nodejs.org/en/blog/release/v14.15.5/ So we can go ahead and merge this @rbuckton

**Comment by voxpelli** [comment]
Rebased on top of master, still needs a review, though I'm not sure @typescript-bot gets that.

**Comment by voxpelli** [comment]
I'll have to look at the CI then... Would be awesome to get some feedback now when the patch has been upstreamed.

**Comment by voxpelli** [comment]
I will try to find time soon

**Pull request closed without merging**